index.js

```diff
@@ -240,8 +246,35 @@ function constructRequest(har, opts = { userAgent: false, files: {}, multipartEn
 
           options.body = JSON.stringify(formBody);
       }
-    } else {
-      options.body = request.postData.text;
+    } else if (request.postData.text.length) {
+      // If we've got `files` map content present, and this post data content contains a valid data URL then we can
+      // substitute the payload with that file instead of the using data URL.
+      if (opts.files) {
+        const parsed = parseDataUrl(request.postData.text);
+        if (parsed && 'name' in parsed && parsed.name in opts.files) {
+          const fileContents = opts.files[parsed.name];
+          if (Buffer.isBuffer(fileContents)) {
+            options.body = fileContents;
+          } else if (isFile(fileContents)) {
+            // `Readable.from` isn't available in browsers but the browser `Request` object can handle `File` objects
+            // just fine without us having to mold it into shape.
+            if (typeof window !== 'undefined' && typeof document !== 'undefined') {
+              options.body = fileContents;
+            } else {
+              options.body = Readable.from(fileContents.stream());
```

Comment:
This is a browser API right? Would this have memory issues on large files? Wonder if there's a way to pass the stream straight through.

Reply:
No, `Readable.from()` is Node-only and even though we're loading in the `readable-stream` package it prevents itself from being used in the browser. That's why we're doing a browser check above here.

There's unfortunately no way with the native `FormData` API to pass a stream straight through to it since streams are a Node thing. The non-compliant `form-data` package supports this, but thats one of the many non-compliant things it does that makes using it in the browser impossible.